aten/src/ATen/native/native_functions.yaml

```diff
@@ -9086,46 +9086,46 @@
 
 - func: histc.out(Tensor self, int bins=100, Scalar min=0, Scalar max=0, *, Tensor(a!) out) -> Tensor(a!)
   dispatch:
-    CPU: histogram_histc_cpu_out
+    CPU, MPS: histogram_histc_out
```

Comment:
@qqaatw , i think we should register this as a separate : `histogram_histc_mps_out` kernel. And not piggy-back on CPU code and make changes in independent layer.

Reply:
`histogram_histc_cpu_out` (and other histc, histogram, and histogramdd functions registered in `native_functions.yaml`) originally lived in `native/Histogram.cpp`, which was not really CPU codes but a general pre-processing and re-dispatching layer in my opinion. The reason it had `cpu` in the name, I think, was lack of the Cuda implementation. (although cuda has `histc` impl, it doesn't make use of `histogramdd` so the code cannot be shared with the CPU)

Furthermore, this layer provides a bunch of input checks and utilities where we don't need to take care of them ourselves. We can of course copy-paste them to mps file and register them separately but it would be hard to maintain (we need to test them ourselves), and be in sync with the other ones.

Reply:
Also for the cpu label in this PR, the reason of it was because I accidentally modified `HistogramKernel.cpp`, the modification was removed.

Reply:
cc @albanD

Reply:
We have a mechanism of registering structured kernels, but they should not contain `is_mps()` calls